### ISSUE TITLE
HASS: libturbojpeg switch to libjpeg-turbo

### ISSUE THREAD

**Issue by H3mul** [onedr0p/containers#1063]
Source:
https://github.com/home-assistant/docker/pull/313

Currently, (using `libturbojpeg`), HASS can't seem to find `libturbojpeg` when using a camera entity:

```
2024-07-15 15:40:01.067 ERROR (MainThread) [homeassistant.components.camera.img_util] Error loading libturbojpeg; Camera snapshot performance will be sub-optimal
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/homeassistant/components/camera/img_util.py", line 98, in __init__
    TurboJPEGSingleton.__instance = TurboJPEG()
                                    ^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/turbojpeg.py", line 298, in __init__
    self.__find_turbojpeg() if lib_path is None else lib_path)
    ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/turbojpeg.py", line 928, in __find_turbojpeg
    raise RuntimeError(
RuntimeError: Unable to locate turbojpeg library automatically. You may specify the turbojpeg library path manually.
e.g. jpeg = TurboJPEG(lib_path)
```

**Comment by onedr0p** [comment]
Already reported here https://github.com/onedr0p/containers/issues/988. As I mentioned I'm not sure what the fix could be.

**Comment by H3mul** [comment]
Sorry for the duplicate, I misunderstood the upstream issue. Thanks for the link

Did a big of digging, [PyTurboJPEG](https://github.com/lilohuang/PyTurboJPEG) just doesn't look in the location of the alpine package.

Short term fix: add `LD_LIBRARY_PATH=/usr/lib` env for HASS container
Long term fix: https://github.com/lilohuang/PyTurboJPEG/pull/79

No fix from this repo required

**Comment by H3mul** [comment]
Small followup - found another alternative while chasing a different lib issue

Alpine linux causes problems with the python [`ctypes.utils.find_library()`](https://docs.python.org/3/library/ctypes.html#ctypes.util.find_library) function, which many packages use. Eg, HASS also can't find `libpcap` for the [same reason](https://github.com/secdev/scapy/blob/4a852fee9114cadccb59fe4e7b742b4b316ce3cc/scapy/libs/winpcapy.py#L38C17-L38C29)

`find_library()` attempts to use [several tools](https://github.com/python/cpython/blob/b455a5a55cb1fd5bb6178a969e8ebd0e6e91b610/Lib/ctypes/util.py#L303), the one that works in Alpine Linux seems to be `ld`, it includes `/usr/lib` by default. But it only finds libraries symlinked to `/usr/lib/*.so`, which [ONLY happens in `-dev` packages](https://stackoverflow.com/questions/69532092/linker-cant-find-external-libraries-on-alpine-like-xcb-and-xkbcommon)

Basically, if we add `binutils` for `ld` and `-dev` versions of lib packages for the symlink, `LD_LIBRARY_PATH=/usr/lib` is not necessary.

quick diff that fixes missing `libturbojpeg` and `libpcap`:

```diff
diff --git a/Dockerfile.onedr0p b/Dockerfile
index 36525de..43d9ca7 100644
--- a/Dockerfile.onedr0p
+++ b/Dockerfile
@@ -24,6 +24,7 @@ WORKDIR /app
 RUN \
     apk add --no-cache \
         bash \
+        binutils \
         bluez \
         bluez-deprecated \
         bluez-libs \
@@ -36,7 +37,8 @@ RUN \
         git \
         jq \
         libcap \
-        libturbojpeg \
+        libjpeg-turbo-dev \
+        libpcap-dev \
         libstdc++ \
         libxslt \
         mariadb-connector-c \
```

**Comment by onedr0p** [comment]
Sounds good... Feel free to open a PR.
